### PR TITLE
Add auto_gptq to lmdeploy lite

### PR DESCRIPTION
**Pull request by AllentDan**
Only supports models that auto-gptq supports.

### CODE REVIEW

**Comment by AllentDan** [comment]
Usage:
```shell
lmdeploy lite auto_gptq InternLM/internlm2-chat-1_8b --work-dir InternLM/internlm2-chat-1_8b-gptq
```

File: lmdeploy/cli/lite.py [hunk 1]
```diff
@@ -97,6 +125,13 @@ def auto_awq(args):
         kwargs = convert_args(args)
         auto_awq(**kwargs)
 
+    @staticmethod
+    def auto_gptq(args):
+        """Perform weight quantization using AWQ algorithm."""
```
Comment (lvhan028) [inline lmdeploy/cli/lite.py:130]: typo "AWQ"

File: lmdeploy/lite/apis/gptq.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            which means only smooth quant with 0.5 ratio will be applied.
+        device (str): Device type of running.
+        revision (str): The specific model version to use. It can be a
+            branch name, a tag name, or a commit id. If unspecified,
+            will use the default version.
+    """
+    try:
+        from auto_gptq import AutoGPTQForCausalLM, BaseQuantizeConfig
+    except Exception:
+        raise ImportError('To use auto_gptq, please install auto-gptq by '
```
Comment (lvhan028) [inline lmdeploy/lite/apis/gptq.py:42]: Great!

**Comment by lvhan028** [comment]
```
The repository for ptb_text_only contains custom code which must be executed to correctly load the dataset. You can inspect the repository content at https://hf.co/datasets/ptb_text_only.
You can avoid this prompt in future by passing the argument `trust_remote_code=True`.

Do you wish to run the custom code? [y/N] y
```
can we hide `trust_remote_code`?

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**